Mathlib/Data/Int/SuccPred.lean

```diff
@@ -36,46 +40,43 @@ theorem succ_eq_succ : Order.succ = succ :=
 theorem pred_eq_pred : Order.pred = pred :=
   rfl
 
+@[deprecated Order.one_le_iff_pos (since := "2024-09-04")]
 theorem pos_iff_one_le {a : ℤ} : 0 < a ↔ 1 ≤ a :=
   Order.succ_le_iff.symm
 
-theorem succ_iterate (a : ℤ) : ∀ n, succ^[n] a = a + n
-  | 0 => (add_zero a).symm
-  | n + 1 => by
-    rw [Function.iterate_succ', Int.ofNat_succ, ← add_assoc]
-    exact congr_arg _ (succ_iterate a n)
+protected theorem succ_iterate (a : ℤ) : ∀ n, succ^[n] a = a + n :=
+  Order.succ_iterate a
 
-theorem pred_iterate (a : ℤ) : ∀ n, pred^[n] a = a - n
-  | 0 => (sub_zero a).symm
-  | n + 1 => by
-    rw [Function.iterate_succ', Int.ofNat_succ, ← sub_sub]
-    exact congr_arg _ (pred_iterate a n)
+protected theorem pred_iterate (a : ℤ) : ∀ n, pred^[n] a = a - n :=
```

Comment:
And this one too

Reply:
I think this one should be kept iff we keep `succ_iterate`.